monai/auto3dseg/algo_gen.py

```diff
@@ -91,6 +91,10 @@ def generate(self):
         """Generate new Algo -- based on data_stats, budget, and history of previous algo generations."""
         pass
 
+    def generate_with_customized_param(self):
+        """Customize parameters for gpu -- based on gpu capacities."""
+        pass
+
```

Comment:
could you make use of the existing `set_budget` API to set the available parameters and use `generate()` to produce the algo?

Reply:
the functionality serves different purpose of `set_budge`. `set_budge` will require input from users to define how much GPU memory auto3dseg can use. but the customization/optimization is trying to fully utilize assigned memory from `set_budge`. It might be better not to mix the concept here.